go/action_kit_sdk/action_sdk.go

```diff
@@ -0,0 +1,429 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2023 Steadybit GmbH
+
+package action_kit_sdk
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/rs/zerolog/log"
+	"github.com/steadybit/action-kit/go/action_kit_api/v2"
+	"github.com/steadybit/action-kit/go/action_kit_sdk/heartbeat"
+	"github.com/steadybit/action-kit/go/action_kit_sdk/state_persister"
+	extension_kit "github.com/steadybit/extension-kit"
+	"github.com/steadybit/extension-kit/extconversion"
+	"github.com/steadybit/extension-kit/exthttp"
+	"net/http"
+	"os"
+	"os/signal"
+	"reflect"
+	"syscall"
+	"time"
+)
+
+var (
+	registeredActions = make(map[string]interface{})
+	statePersister    = state_persister.NewInmemoryStatePersister()
+)
+
+type Action[T any] interface {
+	// NewEmptyState creates a new empty state. A pointer to this state is passed to the other methods.
+	NewEmptyState() T
+	// Describe returns the action description.
+	Describe() action_kit_api.ActionDescription
+	// Prepare is called before the action is actually started. It is used to validate the action configuration and to prepare the action state.
+	// [Details](https://github.com/steadybit/action-kit/blob/main/docs/action-api.md#preparation)
+	Prepare(ctx context.Context, state *T, request action_kit_api.PrepareActionRequestBody) (*action_kit_api.PrepareResult, error)
```

Comment:
I think the signature is now a bit misleading...
so I get passed the state by reference to populate but could also return a PrepareResult which also a state field...
Looks confusing...

So why don't we not use a receiver parameter and use a custom return type for the SDK which accepts any for the state?